src/librustc/ty/layout.rs

```diff
@@ -769,6 +770,7 @@ pub enum Layout {
     CEnum {
         discr: Integer,
         signed: bool,
+        non_zero: bool,
```

Comment:
You already have the range (see `rustc_trans::adt` for a way to determine if a value is in the range), why would this additional field be needed?